calcom/cal.com
# Admin apps UI

## What does this PR do?



This PR allows admins to configure apps for their self-hosted instances. Admins can configure apps in either the setup wizard or the settings pages.

Fixes #5381



**Environment**: Staging(main branch) / Production

## Type of change




## How should this be tested?



- Navigate to "settings/admin/apps/calendar"
- Enable apps
    - If an app does not need a key then it should be enabled automatically
    - If an app needs a key it will still be enabled but the keys should appear under the app
        - Enter the keys for the app and hit saved
        - It should now be saved in the db
    - Enabled apps should only appear in the app store and available apps under an event type

- Navigate to "auth/setup"
- It should bring you to the setup wizard but you won't be able to navigate to any steps
- Either delete all users in the db or remove the boolean checks in "apps/web/pages/auth/setup/index.tsx"
- Create a new admin user. After you should be able to navigate to the admin apps step

Disabling calendar apps
- Add a calendar app
- Set it as your default destination calendar
- Disable the app in the admin settings
- You should receive an email and the calendar should no longer be your destination calendar
- Set a different calendar to a specific calendar
    - When disabled it should change to your default calendar or none will be selected if you do not have any other calendars

Disabling video apps
- Add a video app
- Set it as a location on an event type
- Disable the app in the admin settings
- You should receive an email
- Creating a booking with the event type where you set the disabled video app as a location
- The location should fall back to Cal video

Disabling payment apps
- Add Stripe
- Add a price to an event type
- Disable the app in the admin settings
- You should receive an email with an additional explanation and a link to the affected event type
- Create a booking with that event type. It should not prompt for a payment,


## Checklist



- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
